src/modules/admin/components/formRules/NewFormRuleDialog.tsx

```diff
@@ -0,0 +1,304 @@
+    },
+    onCompleted: (data) => {
+      if (data.createFormRule?.formRule) {
+        onCloseDialog();
+        // clear the cache so that applicability gets updated
+        evictQuery('formDefinition', { id: formId }); // todo @martha - evict specifically projectMatches?
+        evictQuery('formRules');
+      }
+    },
+  });
+
+  const conditions = useMemo(() => {
+    return (
+      [
+        'projectType',
+        'projectId',
+        'organizationId',
+        'funder',
+      ] as ConditionType[]
+    ).map((conditionType) => {
+      return {
+        conditionType: conditionType,
+        value: rule[conditionType],
+      };
+    });
+  }, [rule]);
+
+  const conditionsAvailable: ConditionType[] = useMemo(
+    () =>
+      conditions
+        .filter((condition) => !condition.value)
+        .map((condition) => condition.conditionType),
+    [conditions]
+  );
+
+  const { pickList: projectList } = usePickList({
+    item: {
+      linkId: 'fake',
+      type: ItemType.Choice,
+      pickListReference: PickListType.Project,
+    },
+  });
+
+  const { pickList: orgList } = usePickList({
+    item: {
+      linkId: 'fake',
+      type: ItemType.Choice,
+      pickListReference: PickListType.Organization,
+    },
+  });
+
+  const pickListMap: Record<ConditionType, PickListOption[]> = {
+    projectType: projectTypePickList || [],
+    funder: funderPickList || [],
+    organizationId: orgList || [],
+    projectId: projectList || [],
+  };
+
+  const isTiny = useIsMobile('sm');
+
+  if (error) throw error;
+
+  return (
+    <CommonDialog
+      open={open}
+      fullWidth
+      fullScreen={isTiny}
+      onClose={onCloseDialog}
+    >
+      <DialogTitle>
+        New rule for: <b>{formTitle}</b>
+      </DialogTitle>
+      <DialogContent>
+        <Stack gap={2} direction='row' sx={{ mt: 2, mb: 1, display: 'flex' }}>
+          <FormRuleLabelTypography>Applies to</FormRuleLabelTypography>
+          <FormSelect
+            sx={{ flexGrow: 1 }}
+            value={{
+              code: rule.dataCollectedAbout || DataCollectedAbout.AllClients,
+            }}
+            options={dataCollectedAboutPickList}
+            onChange={(_event, option) => {
+              if (isPickListOption(option)) {
+                setRule({
+                  ...rule,
+                  dataCollectedAbout: option.code as DataCollectedAbout,
+                });
+              }
+            }}
+          />
+        </Stack>
+        <CardGroup
+          onAddItem={() => {
+            if (conditionsAvailable.length > 0) {
+              const conditionType = conditionsAvailable[0];
+              const defaultValue = pickListMap[conditionType][0].code;
+              // We display all conditions with a non-null value, so setting this condition's value
+              // to some default value for its type (e.g. the first Project in the list) causes it to appear.
+              setRule({ ...rule, [conditionType]: defaultValue });
```

Comment:
It seems like using `FormRuleInput` as the internal component state caused some UX limitations:
* the "condition types" are in a fixed order
* requiring us to auto-fill an initial value for the field to appear

The first bullet doesn't seem like a big deal, but I think the second one is worth addressing. I could imagine someone thinking that the selected project is intentional (i.e. "am I editing an existing rule now? did someone already set this to project X? Why does it say Coordinated Entry? etc..) and that causing some support questions/confusion.

The better way to do this might be to refactor so that the component state mimics the interface (e.g. an array of conditions), and it only gets converted into a FormRuleInput on submit. Though that feels like too big of a change to make at this stage, for this component, so I would suggest a different workaround instead:
- set value to an empty string `''` to indicate that it's present. switch to filtering out like `.filter((condition) => typeof condition.value !== 'undefined')`.

Reply:
Thanks -- yes, I had some version of this in the past and would be happy to revert to it.

I'm also happy to spend a few minutes trying to implement your first suggestion. I think it would probably be more readable, and it's probably also a useful exercise for me. I'll timebox my effort to half an hour or so.

Reply:
Sounds good. I'm totally fine with as-is. This might be helpful for your timebox though https://react.dev/learn/updating-arrays-in-state

Reply:
Ok, in a short time box I didn't come up with something that was as functional as / more readable than the code as-is, so I'll leave it!